sycl/source/detail/adapter.hpp

```diff
@@ -71,17 +71,7 @@ class Adapter {
       int32_t adapter_error = 0;
       ur_result = call_nocheck<UrApiKind::urAdapterGetLastError>(
           MAdapter, &message, &adapter_error);
-
-      // If the warning level is greater then 2 emit the message
-      if (message != nullptr &&
-          detail::SYCLConfig<detail::SYCL_RT_WARNING_LEVEL>::get() >= 2) {
-        std::clog << message << std::endl;
-      }
-
-      // If it is a warning do not throw code
-      if (ur_result == UR_RESULT_SUCCESS) {
-        return;
-      }
+      std::cerr << message << " (error code " << adapter_error << ")\n";
```

Comment:
Is there a reason we don't throw instead? It would be more in line with the case below, which we won't hit anymore for adapter-specific failures.